### PR TITLE
Add notifications for all status scenarios and chat messages, sync notifications settings with database and cloud functions, fix some bugs and refactor assigned files based on code review

### LINKED ISSUE #29: As an app user, I want to modify the radius of my notifications, so that I can only be notified about posts in a certain area according to the radius

### LINKED ISSUE #238: As a user I want my settings/preferences to  be saved in the database so that I don't have to change them every time I log in

### LINKED ISSUE #246: I don't want to get the wrong notification when I commit a favor.

### LINKED ISSUE #247: I shouldn't get a notification when I'm the first to chat as a requester.

### LINKED ISSUE #248: In the shop page, it appeared a blank space between the tab page and the toolbar that was not there before

### LINKED ISSUE #258: As a user, I want to be able to turn off notifications for chat messages and favor updates in the settings page

### LINKED ISSUE #260: As a user, I want to receive notifications every time there is some update on a favor I created or committed

### PR DESCRIPTION
**Pull request by mikanikos**
Close #260, close #238, close #29, close #258 

Fix #246, fix #247, fix #248 

Fix some issues outlined in the code review #209 

Remove unused resources and reformat the entire codebase

### CODE REVIEW

File: app/src/main/java/ch/epfl/favo/view/tabs/favorList/FavorPage.java [hunk 1]
```diff
@@ -34,7 +34,7 @@
 import ch.epfl.favo.R;
```
Comment (danielpgp1012) [inline app/src/main/java/ch/epfl/favo/view/tabs/favorList/FavorPage.java:34]: We should rename this class to MyFavorsPage

File: server/functions/index.js [hunk 2]
```diff
@@ -1,19 +1,22 @@
 // The Cloud Functions for Firebase SDK to create Cloud Functions and setup triggers.
```
Comment (danielpgp1012) [inline server/functions/index.js:1]: Can format indents with VsCode

File: server/functions/index.js [hunk 3]
```diff
@@ -263,25 +303,33 @@ exports.expireOldFavorsOnCreate = functions.firestore
...
+    .onCreate((snap) => {
+        const message = snap.data();
+        const senderId = message.uid;
+        const favorId = message.favorId;
 
-        }
+        // find favor with selected favorId
+        return db.collection('/favors').where('id', '==', favorId).get()
+            .then((snapshot) => {
+                snapshot.forEach((doc) => {
```
Comment (danielpgp1012) [inline server/functions/index.js:314]: would a foreach loop here be appropriate given that we're only fetching 1?

File: server/functions/index.js [hunk 4]
```diff
@@ -263,25 +303,33 @@ exports.expireOldFavorsOnCreate = functions.firestore
...
+                        },
+                        notification: {
+                            title: 'New message in the chat of favor ' + favor.title,
+                            body: 'Click to check out the post details',
+                        },
+                        tokens: []
+                    };
+
+                    userIds.splice(userIds.indexOf(senderId), 1)
+                    sendMessageToUsers(notification, userIds, CHAT_TYPE)
```
Comment (danielpgp1012) [inline server/functions/index.js:330]: So right now we send notification to only users who have commited right?

Comment (mikanikos) [reply server/functions/index.js:330]: We send chat notifications to all users who committed and the requester when no user has been accepted. We then send notification to only the accepter and the requester when a user is accepted.

File: server/functions/index.js (hunk 3 above)
Comment (mikanikos) [reply server/functions/index.js:314]: Yeah right, this is also true for the other functions we have (e.g. when retrieving a user based on the id). However I didn't find any other ways to handle query results in the documentation or in other forums (maybe I didn't see or find it). I think this is safe to do and probably not inefficient. However, if anyone knows how to handle this in code in a better way, feel free to suggest (sorry but I don't know Javascript very well).

File: app/src/androidTest/java/ch/epfl/favo/notifications/FirebaseMessagingServiceTest.java [hunk 5]
```diff
@@ -75,7 +75,7 @@ public void testNotifications() {
 
     Bundle bundle = generateBundle();
     FirebaseMessagingService.showNotification(
-        mainActivityTestRule.getActivity(), new RemoteMessage(bundle), "Default channel id");
+        mainActivityTestRule.getActivity(), new RemoteMessage(bundle), FirebaseMessagingService.CHANNEL_NAME);
 
     // WORKS LOCALLY, NOT ON TRAVIS... BUT WORKS ON CIRRUS :)
```
[the remark this answers is not in the document]
Comment (mikanikos) [reply app/src/androidTest/java/ch/epfl/favo/notifications/FirebaseMessagingServiceTest.java]: Done

File: app/src/androidTest/java/ch/epfl/favo/view/SettingsPageTest.java [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+  }
+
+  @Test
+  public void testNearbyFavorNotificationsSwitch() {
+    navigateToSettingsTab();
+
+    // enable switch
+    clickPreference(R.string.notification_title);
+
+//    // set radius
```
[the remark this answers is not in the document]
Comment (mikanikos) [reply app/src/androidTest/java/ch/epfl/favo/view/SettingsPageTest.java]: Done

File: app/src/androidTest/java/ch/epfl/favo/view/SettingsPageTest.java [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+
+  @Test
+  public void testChatNotificationsSwitch() {
+    navigateToSettingsTab();
+
+    // enable switch
+    clickPreference(R.string.notification_chat_title);
+
+    // disable switch
+    clickPreference(R.string.notification_chat_title);
```
[the remark this answers is not in the document]
Comment (mikanikos) [reply app/src/androidTest/java/ch/epfl/favo/view/SettingsPageTest.java:111]: Done

File: app/src/androidTest/java/ch/epfl/favo/view/SettingsPageTest.java [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+
+  @Test
+  public void testUpdateNotificationsSwitch() {
+    navigateToSettingsTab();
+
+    // enable switch
+    clickPreference(R.string.notification_update_title);
+
+    // disable switch
+    clickPreference(R.string.notification_update_title);
```
[the remark this answers is not in the document]
Comment (mikanikos) [reply app/src/androidTest/java/ch/epfl/favo/view/SettingsPageTest.java:124]: Done

File: app/src/main/java/ch/epfl/favo/user/User.java [hunk 9]
```diff
@@ -12,33 +12,35 @@
 import ch.epfl.favo.exception.IllegalAcceptException;
 import ch.epfl.favo.exception.IllegalRequestException;
 import ch.epfl.favo.gps.FavoLocation;
-import ch.epfl.favo.util.CommonTools;
 
 /**
  * This class contains all the relevant information about users TODO: It should implement parcelable
```
[the remark this answers is not in the document]
Comment (mikanikos) [reply app/src/main/java/ch/epfl/favo/user/User.java]: Done

File: app/src/main/java/ch/epfl/favo/view/tabs/favorList/FavorPage.java (hunk 1 above)
Comment (mikanikos) [reply app/src/main/java/ch/epfl/favo/view/tabs/favorList/FavorPage.java:34]: Done

File: server/functions/index.js (hunk 2 above)
Comment (mikanikos) [reply server/functions/index.js:1]: I think it was already formatted using Intellij with the node.js plugin, let me know if you find any inconsistency with the indentation

**Comment by huajian1069** [comment]
Thanks for your efforts!

File: server/functions/index.js (hunk 4 above)
Comment (danielpgp1012) [reply server/functions/index.js:330]: ok great.

File: server/functions/index.js (hunk 3 above)
Comment (danielpgp1012) [reply server/functions/index.js:314]: If no other ways it's not a problem

File: server/functions/index.js [hunk 10]
```diff
@@ -87,13 +114,12 @@ exports.sendNotificationNearbyOnNewFavor = functions.firestore
                     var latUser = user.location.latitude;
                     var longUser = user.location.longitude;
                     var distance = distanceInKm(longFav, latFav, longUser, latUser);
-                    if (distance < maxDistance && user.id !== posterId) {
+                    if (distance < user.notificationRadius && user.id !== posterId && user.activeAcceptingFavors === 0 && user.activeRequestingFavors === 0) {
```
Comment (danielpgp1012) [inline server/functions/index.js:117]: Do you think that those who have requested but haven't accepted should be notified as well? Even If a person has requested a favor, i think it wouldn't be bad to be notified about new favors around

Comment (mikanikos) [reply server/functions/index.js]: I agree, I changed it

**Pull request merged**